pkg/cloud/aws/kinesis/kinesis.go

```diff
@@ -25,9 +28,10 @@ func createKinesisStream(config cfg.Config, logger mon.Logger, client kinesisifa
 	if err != nil {
 		logger.WithFields(mon.Fields{
 			"stream_name": streamName,
-		}).Error(err, "failed to list kinesis streams")
+			"error":       err.Error(),
+		}).Warn("failed to list kinesis streams")
 
-		return
+		return fmt.Errorf("failed to list kinesis streams: %w", err)
```

Comment:
why first logging it as a warning and then returning an error?  
shouldn't the caller log it as a warning if it decides that it is not an error? :thinking: